src/firefly/js/templates/lightcurve/LcPhaseFoldingPopup.jsx

```diff
@@ -0,0 +1,623 @@
+                                return retVal;
+                            },
+                            tooltip: 'time column, mjd, col1, or col2',
+                            label : `${fKeyDef.time.label}:`,
+                            labelWidth : 100
+                        }}
+                    getSuggestions = {(val)=>{
+                            const suggestions = validSuggestions.filter((el)=>{return el.startsWith(val);});
+                            return suggestions.length > 0 ? suggestions : validSuggestions;
+                        }}
+
+                />
+
+                <br/>
+                <ValidationField fieldKey={fKeyDef.flux.fkey}
+                                 forceReinit={true}
+                                 initialState= {{
+                                  fieldKey: fKeyDef.flux.fkey,
+                                  value: '2.0',
+                                  validator: floatValidator.bind(null, 2.0, 5.5, 3,'Flux Column error'),
+                                  tooltip: 'Flux Column, between 2.0 to 5.5',
+                                  label : `${fKeyDef.flux.label}:`,
+                                  labelWidth : 100
+                          }} />
+
+                <br/>
+
+                <ValidationField fieldKey={fKeyDef.fluxerr.fkey}
+                                 forceReinit={true}
+                                 initialState= {{
+                                  fieldKey: fKeyDef.fluxerr.fkey,
+                                  value: '0.02',
+                                  validator: floatValidator.bind(null, 0.01, 0.5, 3,'Flux Error'),
+                                  tooltip: 'Flux Error, between 0.01 to 0.5',
+                                  label : `${fKeyDef.fluxerr.label}:`,
+                                  labelWidth : 100
+                         }} />
+                <br/>
+
+                <ValidationField fieldKey={fKeyDef.tz.fkey}
+                                 forceReinit={true}
+                                 initialState= {{
+                                  fieldKey: fKeyDef.tz.fkey,
+                                  value: `${zeroT}`,
+                                  validator: timezeroValidator.bind(null, 0.0, Number.MAX_VALUE, 8, timeErr),
+                                  tooltip: `time zero, within [${timeZero}, ${timeMax}]`,
+                                  label : `${fKeyDef.tz.label}:`,
+                                  labelWidth : 100
+                         }} />
+                <br/>
+
+                <RangeSlider fieldKey={fKeyDef.period.fkey}
+                             initialState= {{
+                                  fieldKey: fKeyDef.period.fkey,
+                                  value: `${min}`,
+                                  label: `${fKeyDef.period.label}:`,
+                                  validator: periodValidator.bind(null, 0.0, Number.MAX_VALUE, 8, periodErr),
```

Comment:
why does range slider need a validator? I would think it only allows valid values.